### PR TITLE
Auto-merge develop to master after successful PR merges into develop

### LINKED ISSUE #2233: Auto-merge GitHub Action Workflow
## Objective
Add a GitHub Action workflow to auto-merge `develop` into `master` after successful PRs into the `develop` branch.

## Goal
To optimize the PR deployment workflow by removing the need to manually create PRs from `develop` into `master` if the build is green.

## Implementation
1. A GH action to create a PR from `develop` into `master` after any PR is successfully merged into `develop`. Proposed implementation:

```yaml
name: Create PR: develop -> master

on:
  pull_request:
    types: [closed]
    branches:
      - develop

jobs:
    pull-request:
      runs-on: ubuntu-latest
      steps:
      - uses: actions/checkout@master
      - name: pull-request
        uses: repo-sync/pull-request@v2
        if: github.event.pull_request.merged == true
        with:
          github_token: ${{ secrets.GH_TOKEN }}
```

2. A GH action to automatically merge the created PR (^above) into `master`: Proposed implementation:
```yaml
name: Merge PR: develop -> master

on:
  pull_request:
    branches:
      - master

jobs:
  merge-build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - uses: maxkomarychev/merge-pal-action@v0.3.0
		if: github.event.pull_request.head.ref == 'develop'
        with:
          token: ${{ secrets.GH_TOKEN}}
```

Other requirements:
 - The `develop` branch needs protections that require reviews and successful status checks to stop unsuccessful builds from being auto-merged into master.
 - The `master` branch protections must not require reviews for PRs.

### PR DESCRIPTION
**Pull request by amirrustam**
- closes #2233 
- [x] @jennifer-shehane we have to disable `master` branch protections to allow for auto-merging from develop to master
- we have to land this into master so all the new PRs would go through this new workflow

### CODE REVIEW

File: .github/workflows/automerge-pr-develop-master.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+jobs:
+  merge-build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: maxkomarychev/merge-pal-action@v0.3.0
+        if: github.event.pull_request.head.ref == 'develop'
+        with:
+          token: ${{ secrets.GH_TOKEN}}
```
Comment (flotwig) [inline .github/workflows/automerge-pr-develop-master.yml]:
```suggestion
          token: ${{ secrets.GH_TOKEN }}
```

**Comment by flotwig** [comment]
@amirrustam Do you have a link to the sandbox you mentioned testing this in? It looks good from here, but I'm not sure how to validate that it actually works before merging it.

**Comment by amirrustam** [comment]
@flotwig here is the sandbox repo: https://github.com/cypress-io/testing-actions
I did bunch of sanity check scenarios test runs in that repo. It would be appreciated if someone else could do some test runs to ensure all is well.

**Comment by amirrustam** [comment]
@flotwig in that sandbox repo, I have jobs.yml workflow to simulate fake and successful status checks on PRs: https://github.com/cypress-io/testing-actions/blob/master/.github/workflows/jobs.yml

**Pull request merged**